server/sonar-server/src/main/java/org/sonar/server/metric/persistence/MetricDao.java

```diff
@@ -111,6 +111,15 @@ public void insert(DbSession session, MetricDto item, MetricDto... others) {
     return mapper(session).selectAvailableCustomMetricsByComponentUuid(projectKey);
   }
 
+  public List<MetricDto> selectByIds(final DbSession session, List<Integer> ids) {
```

Comment:
this method should take in a Set instead of a list since I reckon the order is irrelevant but the uniqueness of value is important